web/src/beta/lib/core/engines/Cesium/Feature/HeatMap/createMeshImageData.ts

```diff
@@ -0,0 +1,108 @@
+import { quantile } from "d3";
+
+export type MeshImageData = {
+  image: HTMLCanvasElement;
+  width: number;
+  height: number;
+  maxValue?: number;
+  minValue?: number;
+  outlierThreshold?: number;
+};
+
+export async function fetchImageAndCreateMeshImageData(url?: string): Promise<MeshImageData> {
+  if (typeof url !== "string") {
+    console.error("URL for valueMap is undefined");
+    return Promise.reject("Invalid URL");
+  }
+
+  const canvas = document.createElement("canvas");
+  const ctx = canvas.getContext("2d");
+  if (!ctx) {
+    return Promise.reject("Failed to create canvas context");
+  }
+
+  return new Promise((resolve, reject) => {
+    const img = new Image();
+    img.onload = () => {
+      canvas.width = img.width;
+      canvas.height = img.height;
+      ctx.drawImage(img, 0, 0);
+
+      const imageData = ctx.getImageData(0, 0, img.width, img.height).data;
+      const { minValue, maxValue, outlierThreshold } = reverseMeshDataAndComputeValues(
```

Comment:
I think we can skip this process when `maxValue`, `minValue` and `contourSpacing` are provided from `appearance`. Could you improve this?:pray:

Reply:
Done.